src/app/util/af-types.h

```diff
@@ -88,6 +88,37 @@ typedef void (*EmberAfGenericClusterFunction)(void);
  */
 #define EMBER_AF_NULL_MANUFACTURER_CODE 0x0000
 
+/**
+ * @brief Type for align a 16 bits value.
+ *
+ * This structure has the same size as a pointer and is used to align
+ * a 16 bit value to the 16 least significant bits of a pointer.
+ * Note that this structure is aware of the endianess and the pointer size.
+ */
+typedef struct
+{
+#if (BIGENDIAN_CPU)
```

Comment:
could we add checks that BIG_ENDIAN_CPU must be defined and an `#else` branch for the align items to ensure this fails to compile if values are not set as expected?

Reply:
We could use:
#if (defined BIGENDIAN_CPU) && (BIGENDIAN_CPU)
#else
#endif